pkg/registry/core/pod/storage/storage.go

```diff
@@ -221,6 +223,10 @@ func (r *BindingREST) setPodHostAndAnnotations(ctx context.Context, podUID types
 		if pod.Spec.NodeName != "" {
 			return nil, fmt.Errorf("pod %v is already assigned to node %q", pod.Name, pod.Spec.NodeName)
 		}
+		// Reject binding to a scheduling un-ready Pod.
+		if utilfeature.DefaultFeatureGate.Enabled(features.PodSchedulingReadiness) && len(pod.Spec.SchedulingGates) != 0 {
```

Comment:
This should really be in validation - a regular user can patch a pod and set spec.nodeName without binding.

Reply:
ah, update's validation has already been covered. Here it's an additional validation for pod creation - prevent user creating a pod directly with nodeName and non-empty scheduling gates.

Reply:
not quite. L223~L225 is checking whether it's updating nodeName X to Y.

Reply:
> prevent user creating a pod directly with nodeName and non-empty scheduling gates

I see, misunderstood the words here. Great catch.